IdentityCore/src/cache/accessor/MSIDAccountCredentialCache.m

```diff
@@ -556,4 +560,159 @@ - (BOOL)removeAppMetadata:(nonnull MSIDAppMetadataCacheItem *)appMetadata
+    // of the keychain item, avoiding an unnecessary read or update if status is the same
+    MSIDIsFRTEnabledStatus (^checkFeatureFlagsAndReturn)(MSIDIsFRTEnabledStatus) = ^MSIDIsFRTEnabledStatus(MSIDIsFRTEnabledStatus status) {
+        
+        // Check if FRT is enabled by feature flight
+        MSIDFlightManager *flightManager = [MSIDFlightManager sharedInstance];
+        BOOL flagEnableFRT = [flightManager boolForKey:@"enable_client_sfrt_by_tenant_id"]; // TODO: Replace this by the constant from the other branch, and remove the hardcoded YES
+        BOOL flagDisableAllFRT = [flightManager boolForKey:@"disable_client_sfrt_for_all"];
+        BOOL shouldEnableFRT = flagEnableFRT && !flagDisableAllFRT;
+        
+        switch (status)
+        {
+                // No entry in cache
+            case MSIDIsFRTEnabledStatusNotEnabled:
+                if (shouldEnableFRT)
+                {
+                    [self updateFRTSettings:YES context:context error:nil];
+                    return MSIDIsFRTEnabledStatusActive;
+                }
+                break;
+                
+                // Deserialization error, try to enable/disable if needed
+            case MSIDIsFRTEnabledStatusDisabledByDeserializationError:
+                if (shouldEnableFRT)
+                {
+                    [self updateFRTSettings:YES context:context error:nil];
+                    return MSIDIsFRTEnabledStatusActive;
+                }
+                else if (flagDisableAllFRT)
+                {
+                    [self updateFRTSettings:NO context:context error:nil];
+                    return MSIDIsFRTEnabledStatusDisabledByKeychainItem;
+                }
+                break;
+                
+                // FRT is currently enabled, check to see if should be disabled
+            case MSIDIsFRTEnabledStatusActive:
+                if (flagDisableAllFRT)
+                {
+                    [self updateFRTSettings:NO context:context error:nil];
+                    return MSIDIsFRTEnabledStatusDisabledByKeychainItem;
+                }
+                break;
+                
+            // FRT is disabled, check to see if should be enabled
+            case MSIDIsFRTEnabledStatusDisabledByKeychainItem:
+                if (shouldEnableFRT)
+                {
+                    [self updateFRTSettings:YES context:context error:nil];
+                    return MSIDIsFRTEnabledStatusActive;
+                }
+                break;
+                
+                // Error reading keychain item, do not update settings
+            case MSIDIsFRTEnabledStatusDisabledByKeychainError:
+                break;
+                
+                // Feature is disabled by client app, do nothing with keychain item
+            case MSIDIsFRTEnabledStatusDisabledByClientApp:
+                break;
+        }
+        
+        return status;
+    };
+    
+    // Check if FRT is disabled by client
```

Comment:
If FRT is disabled by client, will the old FRT logic keeps working?

Reply:
Yes, that's right.